internal/plugins/helm/v1/init.go

```diff
@@ -160,6 +162,13 @@ func (p *initPlugin) runPhase2() error {
 	if err := scorecard.RunInit(p.config); err != nil {
 		return err
 	}
+
+	if p.doCreateAPI {
+		if err := p.apiPlugin.runPhase2(); err != nil {
```

Comment:
What means phase2? Could we use PostScaffold as we do for Helm one instead?

Reply:
This method was added in #3548. The plugins run in this method are eventually going to be implemented as phase 2 plugins, which run after a phase 1 plugin is done running (after post-scaffold).

Reply:
The plugin interface doesn't have a `PostScaffold` function though right? It is just an implementation detail of how we run the plugin once its called by the Kubebuilder plugin libraries.

We haven't really flushed out Phase 2 plugin design (@hasbro17 is working on that), so its hard to say how exactly it'll work. Suffice to say, the phase 2 plugin will have the full universe of files scaffolded by the phase 1 plugin, and can add, update, or delete files as it pleases (probably).

Reply:
HI @joelanford, 

Sorry, but I think I could not make clear my suggestion and why. Since we are working on in 1.0.0 and it is something that does NOT affect end-users at I pushed the PR: https://github.com/operator-framework/operator-sdk/pull/3658 to make it clear and then we can check it after. 

My concern here is that in IMO it is making the code harder to understand and more complex than it would need to be. 
See that we are using the [RunOptions Interface](https://github.com/operator-framework/operator-sdk/blob/master/internal/kubebuilder/cmdutil/cmdutil.go#L22-L30) when we call the `return cmdutil.Run(p)` but we are not respecting it. 

Note that the SDK "phase 2 plugins" just can be executed after/post the Scaffold. So, in POV shows not make sense we call then in the middle of Scaffold or run funcs besides it is working. Also, by following up the interface and its workflow it shows fits very well in the PostScaffold which makes in the code easier to understand and get maintained.  

Also, why we call the interface if we are not respecting/using it all? See that it is not possible to scaffold the SDK CLI customizations before the plugin scaffold. And then,  why we need the funcs called phase2? Would not the following clear enough for we know what is phase 2 and be able to change it when @hasbro17 provide that? 

```go
func (p *initPlugin) PostScaffold() error {
	// SDK phase 2 plugins.
	if err := manifests.RunInit(p.config); err != nil {
		return err
	}
	if err := scorecard.RunInit(p.config); err != nil {
		return err
	}
	if p.doCreateAPI {
		return p.apiPlugin.Run()
	}
	fmt.Printf("Next: define a resource with:\n$ %s create api\n", p.commandName)
	return nil
}

```

Also, if we move the helm/ansible for their repo all SDK CLI customizations (phase 2) would not be in the CLI since it is part of its domain and not Helm/Ansible as well?

Reply:
HI @joelanford, 

Sorry, but I think I could not make clear my suggestion and why. Since we are working on in 1.0.0 and it is something that does NOT affect end-users at I pushed the PR: https://github.com/operator-framework/operator-sdk/pull/3658 to make it clear and then we can check it after.